### DISCUSSION TITLE
Не хватает оперативной памяти для обработки экспорта временной таблицы в таблицу

### DISCUSSION THREAD

**Discussion by Vadim017** [totumonline/totum-ru-issues-and-discussions#580] [Как это сделать?] [answered]
Помогите решить проблему. С чем может быть связана фатальная ошибка: Allowed memory size of 536870912 bytes exhausted (tried to allocate 262144 bytes) in /home/totum/totum-mit/totum/common/calculates/Calculate.php on line 1230

Из временной таблицы экспортирую данные в обычную таблицу. Из временной таблицы должны экспортироваться 15 столбцов. Количество строчек не важно так как ошибка возникает даже если попытаться передать одну строку таблицы
[пятый вроде рабочий код.txt](https://github.com/user-attachments/files/16427887/default.txt)

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Обычно это бывает где-то в коде/коде действия при добавлении, вы написали что-то вида «взять все, поделить|умножить|еще что-то на все». Но также это банальный зацикл в коде. В вашем случае это он:

<img width="1170" alt="Снимок экрана 2024-07-30 в 14 30 15" src="https://github.com/user-attachments/assets/d851316a-5cac-49a4-afc4-e07d62199147">

И вот я тут немного упростил:

```
a1=: insertList(table: 'zayavki_na_kartridgi'; field: 'nomer_zayavki' = #h_nomer_zayavki; field: 'nomer_zayavki' = $select_list; log: true)
		select_list: selectList(table: $#ntn; field: 'nomer_zayavki')
```

Но возможно вам нужно там много колонок сразу вставлять, тогда:

```
a1=: insertList(table: 'zayavki_na_kartridgi'; field: 'nomer_zayavki' = $selrowlist[[nomer_zayavki]]; field: 'data_podachi_zayavki' = $selrowlist[[data_podachi_zayavki]]; field: 'fio_zayavitelya' = $selrowlist[[fio_zayavitelya]]; field: 'doljnost' = $selrowlist[[doljnost]]; field: 'strukturnoe_podrazdelenie' = $selrowlist[[str_podrazdel]]; field: 'telefon' = $selrowlist[[telefon]]; field: 'email' = $selrowlist[[email]]; field: 'adres' = $selrowlist[[adres]]; field: 'tip_oborudovanija' = #h_tip_oborudovanija; field: 'marka_oborudovanija' = #h_marka; field: 'model_cartridzha' = #h_model_kartr; field: 'inv_nomer' = #h_inv_nomer; field: 'vid_rabot' = #h_vid_rabot; field: 'kolichestvo' = #h_kolichestvo ; field: 'primechanie' = #h_primechanie; field: 'nomer_zayavki' = #h_nomer_zayavki; log: true)

	~selrowlist: selectRowList(table: $#ntn; field: 'nomer_zayavki'; field: 'data_podachi_zayavki'; field: 'fio_zayavitelya'; field: 'doljnost'; field: 'str_podrazdel'; field: 'telefon'; field: 'email'; field: 'adres')
	
```

Я тут всего не знаю, что у вас там где в таблице, но если именование везде правильное, то как-то так

`~` — тильда нужна, что бы зафиксировать строку, те она один раз будет запрошена и все ее последующие вызовы из первой строки не будет повторно ходить в БД. Это важно тк в этом selectrowlist не указана сортировка и БД может при нескольких повторных вызовах вернуть разный порядок.

**Reply by Vadim017** [reply]
Спасибо большое, всё работает!
